### ISSUE TITLE
Unhandled Promise rejection: StaticInjectorError(AppModule)[OidcSecurityService -> Router]

### ISSUE THREAD

**Issue by umfaruki** [damienbod/angular-auth-oidc-client#164]
I am trying to implement this and facing this issue please help:

error is 
![untitled](https://user-images.githubusercontent.com/7438486/35435665-679f3746-02ad-11e8-807c-32944a6c6290.png)

i am using angular 5.2.2

**Comment by damienbod** [comment]
This happens when a required provider is missing

Check you app.module providers.

Greetings Damien

**Comment by umfaruki** [comment]
Hi @damienbod 
After installing @angular/router i start facing this error in my modules
_modules/@angular/router/router"' has no exported member 'Params'

this is my package.json

`"dependencies": {
    "@angular/animations": "^5.2.2",
    "@angular/cdk": "^5.1.0",
    "@angular/common": "^5.2.2",
    "@angular/compiler": "^5.2.2",
    "@angular/core": "^5.2.2",
    "@angular/flex-layout": "^2.0.0-beta.12",
    "@angular/forms": "^5.2.2",
    "@angular/http": "^5.2.2",
    "@angular/material": "^5.1.0",
    "@angular/platform-browser": "^5.2.2",
    "@angular/platform-browser-dynamic": "^5.2.2",
    "@angular/router": "^5.2.2",
    "@ngx-loading-bar/http-client": "^1.1.0",
    "angular-auth-oidc-client": "^4.0.0",
    "angular2-cookie": "^1.2.6",
    "bootstrap": "^4.0.0",
    "core-js": "^2.4.1",
    "font-awesome": "^4.7.0",
    "hammerjs": "^2.0.8",
    "jquery": "^3.3.1",
    "moment": "^2.20.1",
    "ng2-ckeditor": "^1.1.14",
    "ng2-toasty": "^4.0.3",
    "oidc-client": "^1.4.1",
    "primeng": "^5.2.0-rc.2",
    "rxjs": "^5.1.0",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular/cli": "1.6.5",
    "@angular/compiler-cli": "^5.2.2",
    "@angular/language-service": "^5.2.2",
    "@types/jasmine": "~2.8.5",
    "@types/jasminewd2": "~2.0.2",
    "@types/node": "~9.3.0",
    "codelyzer": "~4.1.0",
    "jasmine-core": "~2.9.1",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~2.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.2.2",
    "ts-node": "~4.1.0",
    "tslint": "~5.9.1",
    "typescript": "~2.6.2"
  }`

**Comment by damienbod** [comment]
Looks good with the angular deps, do you receive the well known endpoints from the server? you can check this in the browser F12, network.

Could you also post the app.module?

Greetings Damien

**Comment by sric88** [comment]
I faced similar issue with Service instead of Router, I added my service to the [providers] array in my app.module.ts and also removed map function in my service. 

this may help you. Please check

**Comment by AlGantori** [comment]
If you are using modularized and nested xxxModule.ts make sure you are importing xxxModule.ts in your root app.module.ts
something like this...
```
 imports: [
        BrowserModule,
        HttpModule,
        //
        // My components Imports...
        YTPModule,
        AMSModule,
        UQSModule,
        UMSModule,
        UPSModule,
```

**Comment by FaussYao** [comment]
请检查您报错的OidcSecurityService 文件   [  ****  - > Router] ， 一般这里 *** 是什么， 就是这个文件中 用了router 的写法 ，
但是， 你的appModule中没有配置路由 或 配置了但没有与 此处 *** 这个文件 相关的路由配置；
另一个重点在这里：  No provider for Router?
注：没有配置路由 html 中  写routerLink="."  会报错
       ts文件中构造函数注入了Router等，并使用，但并没有路由， 会报错

**Comment by Booster2ooo** [comment]
I faced a similar problem building a test app for my library (which uses angular-auth-oidc-client).

I read your comments and I must admit I've been confused by those application "providers", I tried to add the Router directly in the list, it resulted in even more DI problems.

I took another approach and realized I've never setup any routing. My solution was to import the RouterModule in the app:

```javascript
import { RouterModule } from '@angular/router';
// ...
@NgModule({
  // ...
  imports: [
    // ...
    RouterModule.forRoot([])
  ]
  // ...
})
export class AppModule { }
```
